modules/azure/compute.go

```diff
@@ -8,97 +8,451 @@ import (
+// This function would fail the test if there is an error.
+func GetVirtualMachineManagedDisks(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	diskNames, err := GetVirtualMachineManagedDisksE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return diskNames
+}
+
+// GetVirtualMachineManagedDisksE gets the list of Managed Disk names of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDisksE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Attach authorizer to the client
-	vmClient.Authorizer = *authorizer
+	// Get VM attached Disks
+	vmDisks := *vm.StorageProfile.DataDisks
+
+	// No Attached Disks present
+	if len(vmDisks) == 0 {
+		return nil, nil
+	}
+
+	// Get the Names of the attached Managed Disks
+	diskNames := make([]string, len(vmDisks))
+	for _, v := range vmDisks {
+		diskNames = append(diskNames, *v.Name)
+	}
 
-	return &vmClient, nil
+	return diskNames, nil
 }
 
-// GetSizeOfVirtualMachine gets the size type of the given Azure Virtual Machine.
+// GetVirtualMachineManagedDiskCount gets the Managed Disk count of the specified Azure Virtual Machine.
 // This function would fail the test if there is an error.
-func GetSizeOfVirtualMachine(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) compute.VirtualMachineSizeTypes {
-	size, err := GetSizeOfVirtualMachineE(t, vmName, resGroupName, subscriptionID)
+func GetVirtualMachineManagedDiskCount(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) int {
+	mngDiskCount, err := GetVirtualMachineManagedDiskCountE(vmName, resGroupName, subscriptionID)
 	require.NoError(t, err)
 
-	return size
+	return mngDiskCount
 }
 
-// GetSizeOfVirtualMachineE gets the size type of the given Azure Virtual Machine
-func GetSizeOfVirtualMachineE(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) (compute.VirtualMachineSizeTypes, error) {
-	// Validate resource group name and subscription ID
-	resGroupName, err := getTargetAzureResourceGroupName(resGroupName)
+// GetVirtualMachineManagedDiskCountE gets the Managed Disk count of the specified Azure Virtual Machine.
+func GetVirtualMachineManagedDiskCountE(vmName string, resGroupName string, subscriptionID string) (int, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return -1, err
+	}
+
+	return len(*vm.StorageProfile.DataDisks), nil
+}
+
+// GetVirtualMachineOsDiskName gets the OS Disk name of the specified Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineOsDiskName(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) string {
```

Comment:
NIT: Go convention is to all caps on acronyms, so this should be `GetVirtualMachineOSDiskName`

Reply:
Updated